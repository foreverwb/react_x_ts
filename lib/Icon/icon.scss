@keyframes ui-keyframes-spin {
  0% {transform: rotate(0);}
  100% {transform: rotate(360deg);}
}
@mixin spin() {
  animation: ui-keyframes-spin 1s linear infinite;
}
.ui-icon {
  width: 1.6em;
  height: 1.6em;
  display: inline-block;
  fill: currentColor;
  &-loading {
    @include spin;
  }
}
