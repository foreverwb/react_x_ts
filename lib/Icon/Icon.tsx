import React from 'react';
import {classes} from '../untils/classes'
import './importAll'
import './icon.scss'

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
  // onClick?: React.MouseEventHandler<SVGElement>
}

const Icon: React.FunctionComponent<IconProps> = (props) => {
  const { className, name, ...restProps } = props;
  return (
      <svg
        className={classes('ui-icon', className)}
        {...restProps}>
        <use xlinkHref={`#${name}`}></use>
      </svg>
  )
};
export default Icon;