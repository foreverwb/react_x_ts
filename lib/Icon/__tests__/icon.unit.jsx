import React from 'react';
import Icon from '../Icon';
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';

describe('Icon', () => {
  it('is svg', () => {
    const json = renderer.create(<Icon name="wechat"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('onClick', () => {
    // let n = 1
    // const func = () => {
    //   n = 2
    // }
    const func = jest.fn()
    const c = mount(<Icon name="qq" onClick={func}/>)
    c.find('svg').simulate('click')
    expect(func).toBeCalled()
  })
})