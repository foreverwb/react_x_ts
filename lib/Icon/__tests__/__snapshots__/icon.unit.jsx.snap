// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon is svg 1`] = `
<svg
  className="ui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
